src/client/mqttstate.rs

```diff
@@ -157,6 +164,13 @@ impl MqttState {
         self.outgoing_pub.len()
     }
 
+    pub fn is_disconnecting(&self) -> bool {
```

Comment:
Instead of adding an explicit method for status, we could just create `mqtt_status()` method to return current `MqttConnectionStatus` and use that in `connection.rs`